ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -133,7 +133,11 @@ public SchemaKStream buildStream(
     );
 
     final KsqlStructuredDataOutputNode noRowKey = outputNodeBuilder.build();
-    createSinkTopic(noRowKey.getKafkaTopicName(), ksqlConfig, kafkaTopicClient);
+    createSinkTopic(noRowKey.getKafkaTopicName(),
+                    ksqlConfig,
+                    kafkaTopicClient,
+                    ((result instanceof SchemaKTable)
+                     && !((SchemaKTable) result).isWindowed()));
```

Comment:
I would move this boolean to a helper method named `shouldCompact`. Makes the intent much clearer.